pysr/ssr.py

```diff
@@ -0,0 +1,171 @@
+import warnings
+from typing import List, Optional, Union
+
+import numpy as np
+
+from .sr import PySRRegressor
+from .utils import ArrayLike
+
+
+class PySRSequenceRegressor(PySRRegressor):
+    def __init__(
+        self,
+        recursive_history_length: int = 0,
+        **kwargs,
+    ):
+        self.recursive_history_length = recursive_history_length
+        super().__init__(**kwargs)
+
+    def fit(
+        self,
+        X,
+        y=None,
+        Xresampled=None,
+        weights=None,
+        variable_names: Optional[ArrayLike[str]] = None,
+        complexity_of_variables: Optional[
+            Union[int, float, List[Union[int, float]]]
+        ] = None,
+        X_units: Optional[ArrayLike[str]] = None,
+        y_units=None,
```

Comment:
```suggestion
        self,
        X,
        *,
        weights=None,
        variable_names: Optional[ArrayLike[str]] = None,
        complexity_of_variables: Optional[
            Union[int, float, List[Union[int, float]]]
        ] = None,
        X_units: Optional[ArrayLike[str]] = None,
```

(This `*` will be nice here for preventing potential errors.)

Reply:
mypy doesn't like it :(
but I will use that in future projects 👍